adapter/internal/discovery/xds/server.go

```diff
@@ -466,6 +478,110 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, deployedEnvironments []*
 		orgIDOpenAPIRoutesMap[organizationID] = routesMap
 	}
 
+	apiName := mgwSwagger.GetTitle()
+	apiVersion := mgwSwagger.GetVersion()
+	apiRangeIdentifier := GenerateIdentifierForAPIWithoutVersion(vHost, apiName)
+
+	isIntelligentRoutingEnabled := conf.Adapter.IsIntelligentRoutingEnabled
+	if isIntelligentRoutingEnabled {
+		apiSemVersion, err := ValidateAndGetVersionComponents(apiVersion)
+		if err == nil && apiSemVersion != nil {
+			// Check the major and minor version ranges of the current API
+			existingMajorRangeLatestSemVersion, isMajorRangeRegexAvailable :=
+				orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier][GetMajorVersionRange(*apiSemVersion)]
+			existingMinorRangeLatestSemVersion, isMinorRangeRegexAvailable :=
+				orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier][GetMinorVersionRange(*apiSemVersion)]
+
+			// Check whether the current API is the latest version in the major and minor version ranges
+			isLatestMajorVersion := !isMajorRangeRegexAvailable || CompareSemanticVersions(existingMajorRangeLatestSemVersion, *apiSemVersion)
+			isLatestMinorVersion := !isMinorRangeRegexAvailable || CompareSemanticVersions(existingMinorRangeLatestSemVersion, *apiSemVersion)
+
+			// Remove the existing regexes from the path specifier when latest major and/or minor version is available
+			if (isMajorRangeRegexAvailable || isMinorRangeRegexAvailable) && (isLatestMajorVersion || isLatestMinorVersion) {
+				// Organization's all apis
+				for apiUUID, swagger := range orgIDAPIMgwSwaggerMap[organizationID] {
+					// API's all versions in the same vHost
+					if swagger.GetTitle() == apiName && swagger.GetVHost() == vHost {
+						if isMajorRangeRegexAvailable && swagger.GetVersion() == existingMajorRangeLatestSemVersion.Version ||
```

Comment:
Can we add the 505 `if` conditions to the to the line 504 if conditions ?
or else what if we move the condition checking to a separate method ?